integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkScalaContainerTest.java

```diff
@@ -0,0 +1,162 @@
+    try {
+      if (spark != null) spark.stop();
+    } catch (Exception e) {
+      logger.error("Unable to shut down pyspark container", e);
+    }
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    try {
+      openLineageClientMockContainer.stop();
+    } catch (Exception e) {
+      logger.error("Unable to shut down openlineage client container", e);
+    }
+    network.close();
+  }
+
+  private GenericContainer createSparkContainer(String script) {
+    return new GenericContainer<>(
+            DockerImageName.parse("bitnami/spark:" + System.getProperty("spark.version")))
+        .withNetwork(network)
+        .withNetworkAliases("spark")
+        .withFileSystemBind("src/test/resources/spark_scala_scripts", "/opt/spark_scala_scripts")
+        .withFileSystemBind("src/test/resources/log4j.properties", "/opt/log4j.properties")
+        .withFileSystemBind("build/libs", "/opt/libs")
+        .withLogConsumer(SparkContainerUtils::consumeOutput)
+        .waitingFor(Wait.forLogMessage(".*scala> :quit.*", 1))
+        .withStartupTimeout(Duration.of(10, ChronoUnit.MINUTES))
+        .dependsOn(openLineageClientMockContainer)
+        .withReuse(true)
+        .withCommand(
+            sparkShellCommandForScript("/opt/spark_scala_scripts/" + script)
+                .toArray(new String[] {}));
+  }
+
+  private List<String> sparkShellCommandForScript(String script) {
+    List<String> command = new ArrayList<>();
+    addSparkConfig(command, "spark.openlineage.transport.type=http");
+    addSparkConfig(
+        command,
+        "spark.openlineage.transport.url=http://openlineageclient:1080/api/v1/namespaces/scala-test");
+    addSparkConfig(command, "spark.openlineage.debugFacet=enabled");
+    addSparkConfig(command, "spark.extraListeners=" + OpenLineageSparkListener.class.getName());
+    addSparkConfig(command, "spark.sql.warehouse.dir=/tmp/warehouse");
+    addSparkConfig(command, "spark.sql.shuffle.partitions=1");
+    addSparkConfig(command, "spark.driver.extraJavaOptions=-Dderby.system.home=/tmp/derby");
+    addSparkConfig(command, "spark.sql.warehouse.dir=/tmp/warehouse");
+    addSparkConfig(command, "spark.jars.ivy=/tmp/.ivy2/");
+    addSparkConfig(command, "spark.openlineage.facets.disabled=");
+    addSparkConfig(
+        command, "spark.driver.extraJavaOptions=-Dlog4j.configuration=/opt/log4j.properties");
+
+    List<String> sparkShell =
+        new ArrayList(Arrays.asList("./bin/spark-shell", "--master", "local", "-i", script));
+    sparkShell.addAll(command);
+    sparkShell.addAll(
+        Arrays.asList("--jars", "/opt/libs/" + System.getProperty("openlineage.spark.jar")));
+
+    log.info("Running spark-shell command: ", String.join(" ", sparkShell));
+
+    return sparkShell;
+  }
+
+  @Test
+  void testScalaUnionRddToParquet() {
+    spark = createSparkContainer("rdd_union.scala");
+    spark.start();
+
+    await()
+        .atMost(Duration.ofSeconds(10))
+        .pollInterval(Duration.ofSeconds(2))
```

Comment:
isn't that a bit slow? Will this test take at a minimum 2 secs?

Reply:
switched to 500ms